Provider doesn't handle Harbor project re-creation correctly

**Describe the bug**
When a project has been created by Terraform with a custom tag immutability rules, retention polices and after the project is removed via webUI manually, the provider crashed with the error that immutable tag rule cannot be found, instead of adding the rule alongside with the project.


**To Reproduce**
Deploy harbor project, retention policy and immutable tag rule

`resource "harbor_project" "harbor_project" {
  for_each = {
    for project in local.harbor_projects : project.name => project
  }
  name                        = each.value.name
  public                      = each.value.public
  vulnerability_scanning      = coalesce(each.value.vulnerability_scanning, false)
  enable_content_trust        = each.value.enable_content_trust
  enable_content_trust_cosign = each.value.enable_content_trust_cosign
  auto_sbom_generation        = each.value.auto_sbom_generation
  storage_quota               = each.value.storage_quota

  registry_id = each.value.project_type == "proxy_cache" ? harbor_registry.harbor_registry_main[each.key].registry_id : null

}`

`resource "harbor_retention_policy" "retention_policy" {
  for_each = {
    for project in local.harbor_projects : project.name => project if length(coalesce(project.retention_rules, [])) > 0
  }
  scope    = harbor_project.harbor_project[each.key].id
  schedule = each.value.retention_policy_schedule

  dynamic "rule" {
    for_each = coalesce(each.value.retention_rules, [])

    content {
      always_retain          = rule.value.always_retain
      most_recently_pulled   = rule.value.most_recently_pulled
      most_recently_pushed   = rule.value.most_recently_pushed
      n_days_since_last_pull = rule.value.n_days_since_last_pull
      n_days_since_last_push = rule.value.n_days_since_last_push
      repo_matching          = rule.value.repo_matching
      tag_matching           = rule.value.tag_matching
      repo_excluding         = rule.value.repo_excluding
      tag_excluding          = rule.value.tag_excluding
      untagged_artifacts     = rule.value.untagged_artifacts
    }

  }
  lifecycle {
    replace_triggered_by = [harbor_project.harbor_project[each.key].id]
  }
  depends_on = [harbor_project.harbor_project]
}`

`resource "harbor_immutable_tag_rule" "immutable_tag_rule" {
  for_each = {
    for project in local.harbor_projects : project.name => project if project.project_type != "proxy_cache" && length(coalesce(project.immutable_tag_rules, [])) > 0
  }
  disabled       = false
  project_id     = harbor_project.harbor_project[each.key].id
  repo_excluding = each.value.immutable_tag_rules[0].repo_excluding
  tag_excluding  = each.value.immutable_tag_rules[0].tag_excluding
  repo_matching  = each.value.immutable_tag_rules[0].repo_matching
  tag_matching   = each.value.immutable_tag_rules[0].tag_matching

  lifecycle {
    replace_triggered_by = [harbor_project.harbor_project[each.key].id]
  }

  depends_on = [harbor_project.harbor_project]
}`

Now delete harbor project from webUi manually, on the next run the error occurs

`│ Error: [ERROR] unexpected status code got: 500 expected: 200 
│  {"errors":[{"code":"UNKNOWN","message":"internal server error"}]}
│ 
│ 
│   with harbor_retention_policy.retention_policy["development"],
│   on main.tf line 136, in resource "harbor_retention_policy" "retention_policy":
│  136: resource "harbor_retention_policy" "retention_policy" {
│ 
╵
╷
│ Error: Resource not found /projects/21/immutabletagrules/5
│ 
│   with harbor_immutable_tag_rule.immutable_tag_rule["development"],
│   on main.tf line 163, in resource "harbor_immutable_tag_rule" "immutable_tag_rule":
│  163: resource "harbor_immutable_tag_rule" "immutable_tag_rule" {`

**Expected behavior**
The provider should catch the error and re-create a project with corresponding immutable tag rule and retention policy

**Additional context**

- Provider Version 3.10.15
- Terraform Version 1.9.8
- Harbor Version (harbor-core:v2.11.1, )

--- a/provider/resource_immutable_tag_rule.go
+++ b/provider/resource_immutable_tag_rule.go
@@ -70,7 +70,7 @@ func resourceImmutableTagRuleCreate(d *schema.ResourceData, m interface{}) error
 		return err
 	}
 
-	id, err = client.GetID(headers)
+	id, _ = client.GetID(headers)
 	d.SetId(id)
 	return resourceImmutableTagRuleRead(d, m)
 }
@@ -89,9 +89,12 @@ func resourceImmutableTagRuleRead(d *schema.ResourceData, m interface{}) error {
 	page := 1
 	pageSize := 15
 	for {
-		resp, _, _, err := apiClient.SendRequest("GET", fmt.Sprintf("%s?page=%d&page_size=%d", projectImmutableTagRulePath, page, pageSize), nil, 200)
-		if err != nil {
-			return fmt.Errorf("Resource not found %s", projectImmutableTagRulePath)
+		resp, _, respCode, err := apiClient.SendRequest("GET", fmt.Sprintf("%s?page=%d&page_size=%d", projectImmutableTagRulePath, page, pageSize), nil, 200)
+		if respCode == 404 && err != nil {
+			d.SetId("")
+			return nil
+		} else if err != nil {
+			return fmt.Errorf("resource not found %s / %s / %d", projectImmutableTagRulePath, err, respCode)
 		}
 		var pageModels []models.ImmutableTagRule
 		err = json.Unmarshal([]byte(resp), &pageModels)
@@ -129,7 +132,9 @@ func resourceImmutableTagRuleRead(d *schema.ResourceData, m interface{}) error {
 		}
 	}
 
-	return fmt.Errorf("Resource not found %s", d.Id())
+	log.Printf("[DEBUG] resource not found %s", d.Id())
+	d.SetId("")
+	return nil
 }
 
 func resourceImmutableTagRuleUpdate(d *schema.ResourceData, m interface{}) error {

--- a/provider/resource_retention_policy.go
+++ b/provider/resource_retention_policy.go
@@ -108,6 +108,9 @@ func resourceRetentionCreate(d *schema.ResourceData, m interface{}) error {
 	if err != nil {
 		project_id := strconv.Itoa(body.Scope.Ref)
 		resp, _, _, err := apiClient.SendRequest("GET", models.PathProjects+"/"+project_id, nil, 200)
+		if err != nil {
+			return err
+		}
 
 		var jsonData models.ProjectsBodyResponses
 		err = json.Unmarshal([]byte(resp), &jsonData)
@@ -115,13 +118,13 @@ func resourceRetentionCreate(d *schema.ResourceData, m interface{}) error {
 		if err != nil {
 			return err
 		}
-		_, headers, _, err = apiClient.SendRequest("PUT", models.PathRetentions+"/"+jsonData.Metadata.RetentionId, body, 200)
+		_, _, _, err = apiClient.SendRequest("PUT", models.PathRetentions+"/"+jsonData.Metadata.RetentionId, body, 200)
 		if err != nil {
 			return err
 		}
 		id = models.PathRetentions + "/" + jsonData.Metadata.RetentionId
 	} else {
-		id, err = client.GetID(headers)
+		id, _ = client.GetID(headers)
 	}
 
 	d.SetId(id)
@@ -132,8 +135,11 @@ func resourceRetentionRead(d *schema.ResourceData, m interface{}) error {
 	apiClient := m.(*client.Client)
 
 	log.Printf("[DEBUG] Id: %+v\n", d.Id())
-	resp, _, _, err := apiClient.SendRequest("GET", d.Id(), nil, 200)
-	if err != nil {
+	resp, _, respCode, err := apiClient.SendRequest("GET", d.Id(), nil, 200)
+	if respCode == 404 && err != nil {
+		d.SetId("")
+		return nil
+	} else if err != nil {
 		return err
 	}
 
@@ -181,10 +187,10 @@ func resourceRetentionDelete(d *schema.ResourceData, m interface{}) error {
 	apiClient := m.(*client.Client)
 
 	scope := d.Get("scope").(string)
-	project_id, err := strconv.Atoi(strings.ReplaceAll(scope, "/projects/", ""))
+	project_id, _ := strconv.Atoi(strings.ReplaceAll(scope, "/projects/", ""))
 
 	retention := d.Id()
-	retention_id, err := strconv.Atoi(strings.ReplaceAll(retention, "/retentions/", ""))
+	retention_id, _ := strconv.Atoi(strings.ReplaceAll(retention, "/retentions/", ""))
 
 	body := models.Retention{
 		Algorithm: "or",
@@ -202,7 +208,7 @@ func resourceRetentionDelete(d *schema.ResourceData, m interface{}) error {
 		Id:    retention_id,
 	}
 
-	_, _, _, err = apiClient.SendRequest("PUT", d.Id(), body, 200)
+	_, _, _, err := apiClient.SendRequest("PUT", d.Id(), body, 200)
 	if err != nil {
 		return err
 	}

